modal/client.py

```diff
@@ -366,30 +362,37 @@ async def __call__(
         metadata: Optional[_MetadataLike] = None,
     ) -> ResponseType:
         if self.client._snapshotted:
-            logger.debug(f"refreshing client after snapshot for {self._wrapped_method_name}")
+            logger.debug(f"refreshing client after snapshot for {self.name.rsplit('/', 1)[1]}")
             self.client = await _Client.from_env()
```

Comment:
unrelated to this PR, but:

- `await _Client.from_env()` is called once _for every unique RPC_ that is made after snapshotting
- I don't fully understand why we need to completely recreate the client -- isn't it sufficient to just reestablish connections to the server, similar to what we do when the process forks?

Reply:
For the second bullet - It could have something to do with the auth credentials possibly changing from the snapshot to the resume (iirc I didn't work on this directly, but I have a vague memory of something of this effect being an issue)